core/src/main/java/org/jboss/sbomer/core/enums/SbomStatus.java

```diff
@@ -0,0 +1,25 @@
+/**
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2023 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.sbomer.core.enums;
+
+/**
+ * Enum representing the status of the Sbom.
+ */
+public enum SbomStatus {
+    NEW, GENERATING, GENERATION_FAILED, PROCESSING, PROCESSING_FAILED, READY;
```

Comment:
This is pretty much up for a discussion, especially in case we would like to add multiple processing phases.

Reply:
The `*_FAILED` are not used yet. The reason for it is that we are not querying the TaskRun for status, nor we are receiving events from these which could led to updating the status. This will be changed at some point: https://github.com/project-ncl/sbomer/issues/41

Reply:
In my local work I changed this to: `NEW, IN_PROGRESS, FAILED, READY;` This fits much better with the work I do for updating the status based on the TaskRun outcome.

Reply:
Ok so `NEW` is as soon as a generation is requested, while `IN_PROGRESS`, `FAILED`, `READY` can refer both to generation and processing tasks, so e.g. to get a successful generated SBOM I would need to query for SBOMs with status = `READY` and generator != null and processor == null, while to get a successful processed SBOM I would need to query for SBOMs with status = `READY` and generator != null and processor != null ?

Reply:
Exactly what you said!

Reply:
With one small subtle change: `NEW` is both, for processing and generation. It's when the resource is created, but *not yet* worked on. In other words: after creation but before the TaskRun starts.

Reply:
ok cool, it makes it easier now. in case we will need multiple processing steps we might need to adjust, but many places in the code would need to change, so this would be a minor change. thanks!